libsolidity/analysis/TypeChecker.cpp

```diff
@@ -363,6 +363,35 @@ void TypeChecker::checkLibraryRequirements(ContractDefinition const& _contract)
 			m_errorReporter.typeError(var->location(), "Library cannot have non-constant state variables");
 }
 
+void TypeChecker::checkDoubleStorageAssignment(Assignment const& _assignment)
+{
+	TupleType const& lhs = *dynamic_cast<TupleType const*>(type(_assignment.leftHandSide()).get());
```

Comment:
Oh sorry, this should either be a different dynamic cast, or an assertion, right.